acme/acme/challenges.py

```diff
@@ -479,13 +484,15 @@ def verify_cert(self, domain: str, cert: crypto.X509) -> bool:
         if len(names) != 1 or names[0].lower() != domain.lower():
             return False
 
-        for i in range(cert.get_extension_count()):
-            ext = cert.get_extension(i)
-            # FIXME: assume this is the ACME extension. Currently there is no
-            # way to get full OID of an unknown extension from pyopenssl.
-            if ext.get_short_name() == b'UNDEF':
-                data = ext.get_data()
-                return data == self.h
+        crypto_cert = cert.to_cryptography()
```

Comment:
I wonder if it's worth it to fully switch to using cryptography instead of `crypto` while we're here (making breaking changes)? The note at the top of pyopenssl's [crypto docs page](https://www.pyopenssl.org/en/latest/api/crypto.html) makes it seem like it might all go away at some point, even if it isn't deprecated yet. Especially since we immediately convert to `cryptography` internally in this method. But maybe not worth completely removing it now, especially all at once. I know this would be a lot more work, really just thinking out loud here.

Reply:
i definitely think we ought to switch to cryptography, but the goal here was definitely to avoid breaking changes. once we do a major version change and deprecate the older functions, that might be a good time to do the big switcheroo

Reply:
We do want to remove most of the non-SSL functionality from pyOpenSSL: https://github.com/pyca/pyopenssl/issues/1321 tracks this work